server/src/clang-utils/SourceToHeaderMatchCallback.cpp

```diff
@@ -299,6 +299,12 @@ std::string SourceToHeaderMatchCallback::decorate(std::string_view name) const {
     return forStubHeader ? std::string(name) : NameDecorator::decorate(name);
 }
 
+static bool beginsWithFloat(const std::string& str) {
+    if (str.length() >= 6 && str.substr(0, 6) == "_Float") {
```

Comment:
You shouldn't check that length of string >= 6. 
`If the requested substring extends past the end of the string, i.e. the count is greater than size() - pos (e.g. if count == npos), the returned substring is [pos, size())` 
(https://en.cppreference.com/w/cpp/string/basic_string/size).
You should only chech that string is not empty (but maybe you are sure that string is not empty?...)